src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/ReferenceHandlerTests.IgnoreCycles.cs

```diff
@@ -342,6 +342,41 @@ public async Task IgnoreCycles_WhenWritingNull()
             await Test_Serialize_And_SerializeAsync(rootDictionary, @"{""self"":null}", opts);
         }
 
+        [Fact] // https://github.com/dotnet/runtime/issues/51837
+        public async void IgnoreCycles_StringShouldNotBeIgnored()
+        {
+            var root = new Person
+            {
+                Name = "John",
+                Parent = new Person
+                {
+                    Name = "John",
```

Comment:
Doesn't change anything, but it helps to better emphasize what this test is meant to check:
```suggestion
            var stringReference = "John";
            
            var root = new Person
            {
                Name = stringReference,
                Parent = new Person
                {
                    Name = stringReference,
```